src/probnum/diffeq/odefiltsmooth/initialize.py

```diff
@@ -0,0 +1,318 @@
+"""Initialisation procedures."""
+# pylint: disable=import-outside-toplevel
+
+
+import numpy as np
+import scipy.integrate as sci
+
+import probnum.filtsmooth as pnfs
+import probnum.random_variables as pnrv
+from probnum.filtsmooth import statespace as pnss
+from probnum.problems import InitialValueProblem
+
+__all__ = [
+    "extend_ivp_with_all_derivatives",
+    "compute_all_derivatives_via_rk",
+    "compute_all_derivatives_via_taylormode",
+]
+
+# pylint: disable=import-outside-toplevel
+
+
+def extend_ivp_with_all_derivatives(ivp, order):
+    r"""Create a new InitialValueProblem which is informed about derivatives of initial conditions.
+
+    If the InitialValueProblem is compatible with JAX (and if jax and jaxlib can be imported), this is done with
+    Taylor-mode automatic differentiation.
+    If not, an integrated Brownian motion process is fitted to the first few states of an ODE solution computed with a Runge-Kutta formula.
+
+
+
+    Parameters
+    ----------
+    ivp: InitialValueProblem
+        Initial value problem. See `probnum.problems`.
+    order : int
+        How many derivatives are required? Optional, default is 6. This will later be referred to as :math:`\nu`.
+
+    Raises
+    ------
+    ImportError
+        If `jax` or `jaxlib` are not available.
+
+    Returns
+    -------
+    InitialValueProblem
+        InitialValueProblem object where the attribute `dy0_all` is filled with an
+        :math:`d(\nu + 1)` vector, where :math:`\nu` is the specified order,
+        and :math:`d` is the dimension of the IVP.
+
+
+    Examples
+    --------
+    >>> import sys, pytest
+    >>> if sys.platform.startswith('win'):
+    ...     pytest.skip('this doctest does not work on Windows')
```

Comment:
pytest checks the doctests, but this doctest here does not work on windows (bc. of jax).